raster/r.surf.random/main.c

```diff
@@ -97,6 +101,21 @@ int main(int argc, char *argv[])
     max->type = TYPE_DOUBLE;
     max->answer = "100";
 
+    seed = G_define_option();
+    seed->key = "seed";
+    seed->type = TYPE_INTEGER;
+    seed->required = NO;
+    seed->label = _("Seed for random number generator");
+    seed->description = _("The same seed can be used to obtain same results"
+                          " or random seed can be generated by other means.");
+
+    s_flag = G_define_flag();
```

Comment:
Why this flag? The seed option is not set as a mandatary one thus it is possible to simply drop the flag and generate random seed if no value is passed to the seed option.

Reply:
I was following the logic used in other modules. Where if the the seed flag and seed  option are both set the random seed will be used instead of the user defined seed value. If neither the seed flag or seed option is set the default behavior is to generate a random seed and give the user a warning to set the seed flag or seed option.

We can figure out if we want to change this logic structure when we introduce `G_OPT_M_SEED`.

Reply:
Unfortunately we can not decide it with introduction of G_OPT_M_SEED. If we add a flag or option, it has to stay till the next major release as there can be users / scripts depending on the new flag / option.

You have a good point that other modules do so. We'll have to look into this before pushing new seed flags / options for other modules.

Reply:
The current approach is what r.mapcalc has. `-s` or `seed` is required when `rand()` is used. In most modules, random numbers are always needed, so one of `-s` or `seed` is always required. This can be introduced in a backward compatible way by adding the flag only as optional, but also behaving the same when the flag is not used assuming that that was the previous behavior. (Some modules simply used a fixed seed, so the legacy behavior would be the default.)

Reply:
I found the original discussion in 2014 (using rand(x,y) in r.mapcalc) https://lists.osgeo.org/pipermail/grass-dev/2014-July/069886.html

The intent was to enforce reproducibility (although that makes GRASS to differ from R, Python). As either -s or seed= should be mandatory, a comment on defaulting to -s should be added as we can change that behaviour only for G9. Or should we already create an issue in the tracker with target milestone set to 9.0 and pointing to these PRs?

Reply:
Sensu, if we keep consistency among all modules, +1 from me.